### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

File: src/derived/slot.rs [hunk 1]
```diff
@@ -165,13 +166,13 @@ where
...
-        // can sometimes encounter deadlocks.
-        let old_memo = match self.probe(db, self.state.write(), runtime, revision_now) {
+        let old_memo = match self.probe(db, self.state.upgradable_read(), runtime, revision_now) {
             ProbeState::UpToDate(v) => return v,
-            ProbeState::StaleOrAbsent(mut state) => {
+            ProbeState::StaleOrAbsent(state) => {
+                type RwLockUpgradableReadGuard<'a, T> =
+                    lock_api::RwLockUpgradableReadGuard<'a, RawRwLock, T>;
+
+                let mut state = RwLockUpgradableReadGuard::upgrade(state);
```
Comment (lnicola) [inline src/derived/slot.rs:175]: This looks weird, but I couldn't figure out how to call this otherwise.

Comment (matklad) [reply src/derived/slot.rs:175]: Let's maybe just keep `.write` here, to not mix up library update with changes?

Comment (lnicola) [reply src/derived/slot.rs]: Done.

**Comment by matklad** [comment]
released as 0.14.2

**Pull request merged**